src/main/java/com/minecolonies/entity/EntityCitizen.java

```diff
@@ -475,11 +510,20 @@ public void onDeath(DamageSource par1DamageSource)
 
         if (colony != null)
         {
-            LanguageHandler.sendPlayersLocalizedMessage(
-              colony.getMessageEntityPlayers(),
-              "tile.blockHutTownHall.messageColonistDead",
-              citizenData.getName());
-
+            if(getColonyJob() != null && getColonyJob() instanceof JobGuard)
+            {
+                LanguageHandler.sendPlayersLocalizedMessage(
+                        colony.getMessageEntityPlayers(),
+                        "tile.blockHutTownHall.messageGuardDead",
+                        citizenData.getName());
+            }
+            else
```

Comment:
I suggest we make each job a key for translation and add a `String getNameKey` to return the key and then translate on the client